pkg/networkserver/grpc_gsns.go

```diff
@@ -437,23 +443,30 @@ func (ns *NetworkServer) handleUplink(ctx context.Context, up *ttnpb.UplinkMessa
 	for r := bytes.NewReader(mac); r.Len() > 0; {
 		cmd := &ttnpb.MACCommand{}
 		if err := lorawan.DefaultMACCommands.ReadUplink(r, cmd); err != nil {
-			logger.
-				WithField("unmarshaled", len(cmds)).
-				WithError(err).
-				Warn("Failed to unmarshal MAC commands")
+			logger.WithFields(log.Fields(
+				"bytes_left", r.Len(),
+				"mac_count", len(cmds),
+			)).WithError(err).Warn("Failed to unmarshal MAC command")
 			break
 		}
+		logger.WithField("cid", cmd.CID).Debug("Read MAC command")
```

Comment:
This is going to generate multiple log messages for each LoRaWAN message. Are we sure we want this?